src/main/java/seedu/duke/task/TaskList.java

```diff
@@ -0,0 +1,155 @@
+package seedu.duke.task;
+
+import seedu.duke.chatbot.Ui;
+import seedu.duke.exceptions.DukeException;
+import seedu.duke.exceptions.NoValidTaskIndexException;
+import seedu.duke.exceptions.TaskAlreadyMarkedException;
+
+import java.util.ArrayList;
+
+/**
+ * TaskList is used to track all tasks taken in by Duke.
+ */
+public class TaskList {
+    /**
+     *An array that contains all tasks that will be added.
+     */
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList with no parameters.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Creates a {@link TaskList} with attribute tasks being specified as a given {@link ArrayList}.
+     * @param newTaskList is an {@link ArrayList} of {@link Task}.
+     */
+    TaskList(ArrayList<Task> newTaskList) {
+        this.tasks = newTaskList;
+    }
+
+    /**
+     * returns the number of tasks in the {@link ArrayList} of {@link Task}.
+     * @return an integer specifying the number of tasks
+     */
+    public int getNumberOfTasks() {
+        return this.tasks.size();
+    }
+
+    public ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Used to print all tasks when command "list" is called in Duke.
+     */
+    public void printTasks() {
```

Comment:
Perhaps you could consider using ` for (Task t: tasks) { ... } ` to traverse over the ArrayList<Task>? I'm not sure if it is fully applicable here, but I found it useful and a little more concise for my own method bodies! :)